test/e2e/ginkgo_operator-sdk.go

```diff
@@ -0,0 +1,230 @@
+	if ginkgoNode == 1 {
+		// write the file
+		files, err := ioutil.ReadDir(rootDir + "/deploy/crds")
+		if err != nil {
+			Logf("Failed to read directory %s %s", rootDir+"/deploy/crds", err.Error())
+			return err
+		}
+
+		combined := []byte{}
+		for _, file := range files {
+			bytes, err := ioutil.ReadFile(rootDir + "/deploy/crds/" + file.Name())
+			if err != nil {
+				Logf("Failed to read file %s %s", rootDir+"/deploy/crds/"+file.Name(), err.Error())
+				return err
+			}
+			combined = append(combined, bytes...)
+		}
+
+		err = ioutil.WriteFile(globalManFile, combined, 0644)
+		if err != nil {
+			Logf("Failed to write file %s %s", globalManFile, err.Error())
+			return err
+		}
+	}
+
+	// specify the file name for namespace resource manifest
+	namespacedManFile := fmt.Sprintf("%s/namespace-manifest-%s.yaml", os.TempDir(), ginkgoSeed)
+	if ginkgoNode == 1 {
+		// write the file (empty)
+		err = ioutil.WriteFile(namespacedManFile, []byte{}, 0644)
+		if err != nil {
+			Logf("Failed to write file %s %s", namespacedManFile, err.Error())
+			return err
+		}
+	} else {
+		// ensure the file exists
+		for _, err = os.Stat(namespacedManFile); err != nil && os.IsNotExist(err); {
+			time.Sleep(100 * time.Millisecond)
+			_, err = os.Stat(namespacedManFile)
+		}
+	}
+
+	// specify the args
+	os.Args = append(os.Args, "-root", rootDir, "-singleNamespace", "-globalMan", globalManFile, "-namespacedMan", namespacedManFile)
+
+	return nil
+}
+
+func cleanupOperatorSDK() error {
+	// determine the ginkgo seed
+	ginkgoSeed, err := getGinkoSeed()
+	if err != nil {
+		return err
+	}
+
+	globalManFile := fmt.Sprintf("%s/global-manifest-%s.yaml", os.TempDir(), ginkgoSeed)
+	err = os.Remove(globalManFile)
+	if err != nil {
+		return err
+	}
+
+	namespacedManFile := fmt.Sprintf("%s/namespace-manifest-%s.yaml", os.TempDir(), ginkgoSeed)
+	err = os.Remove(namespacedManFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getArg(argName string) (bool, string) {
+	for i, arg := range os.Args {
+		if arg == argName {
+			return true, os.Args[i+1]
+		} else if strings.HasPrefix(arg, argName+"=") {
+			argAndValue := strings.SplitN(arg, "=", 2)
+			return true, argAndValue[1]
+		}
+	}
+	return false, ""
+}
+
+func getGinkgoNode() int {
+	defined, ginkgoNodeString := getArg("--ginkgo.parallel.node")
+	if !defined {
+		return 1
+	}
+	ginkgoNode, err := strconv.Atoi(ginkgoNodeString)
+	if err != nil {
+		fmt.Printf("Error: %s", err.Error())
+		return 0
+	}
+	return ginkgoNode
+}
+
+func getGinkoSeed() (string, error) {
+	defined, ginkgoSeed := getArg("--ginkgo.seed")
+	if !defined {
+		return "", errors.New("unable to find --ginko.seed argument")
+	}
+	return ginkgoSeed, nil
+}
+
+func getRootDir() (string, error) {
+	rootDir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	for info, err := os.Stat(rootDir + "/LICENSE"); os.IsNotExist(err) || (err == nil && info.IsDir()); {
```

Comment:
Can you please be more specific = what is `dir` referring to?

If remember correctly, then I think I did this because `os.Getwd()` was the directory where ginkgo was invoked for node 1 vs the e2e directory for all other nodes. But I could imagine there is some better utility somewhere, maybe from ginkgo to always get the test directory.

Reply:
Discussed with @qu1queee; will check the behavior and improve at least with code comments.

Reply:
Done. Added comment. See https://github.com/redhat-developer/build/pull/163/commits/234f0c61bda2936f80acb69ef2f0ad37a152d2fc